### PR TITLE
[rush] Improve schema for build-cache.json

### PR DESCRIPTION
**Pull request by octogonz**
## Summary

Working on the build cache website docs motivated some minor improvements to the build-cache.json config file.

## Details

- It's confusing for CLI parameters to magically appear based on **experiments.json**; now they are always visible
- Since that's all the experiment controlled, I've replaced it with a `"buildCacheEnabled"` setting in **build-cache.json**
- I've added a `rush init` template for that file
- I relaxed the JSON schema so that that **build-cache.json** template doesn't need to comment out unused settings;  generally I think this is a better design, e.g. if someone temporarily switches their cache provider, they shouldn't have to delete all their settings

## How it was tested

TBD

### CODE REVIEW

File: apps/rush-lib/src/cli/actions/UpdateCloudCredentialsAction.ts [hunk 1]
```diff
@@ -63,12 +56,19 @@ export class UpdateCloudCredentialsAction extends BaseRushAction {
         this.rushConfiguration
       );
       terminal.writeErrorLine(
-        `The a build cache has not been configured. Configure it by creating a ` +
+        `The build cache has not been configured. Configure it by creating a ` +
```
Comment (iclanton) [inline apps/rush-lib/src/cli/actions/UpdateCloudCredentialsAction.ts:59]: Is it possible to hit this without having a build-cache.json file?

File: apps/rush-lib/src/cli/actions/UpdateCloudCredentialsAction.ts [hunk 2]
```diff
@@ -63,12 +56,19 @@ export class UpdateCloudCredentialsAction extends BaseRushAction {
...
           `"${buildCacheConfigurationFilePath}" file.`
       );
       throw new AlreadyReportedError();
     }
 
+    if (!buildCacheConfiguration.buildCacheEnabled) {
+      terminal.writeErrorLine(
+        `The buildCache feature has not been enabled in ${RushConstants.experimentsFilename}.`
```
Comment (iclanton) [inline apps/rush-lib/src/cli/actions/UpdateCloudCredentialsAction.ts:67]: Aren't you removing the experiment?

**Review by iclanton — APPROVED** [review]

**Comment by elliot-nelson** [comment]
FWIW, checked out your branch and ran it on my "setup" branches in our local monorepo, and it seems to pass the smell test...

✅ Run with _no_ `build-cache.json`, no unwanted verbose/warning output.
✅ Create an empty `build-cache.json`, running does error since it doesn't meet schema (that's OK -- you'll get prompted to add the `cacheProvider` and `buildCacheEnabled` keys, which is a feature imo).
✅ Create valid `build-cache.json` with `buildCacheEnabled: false`, no unwanted verbose/warning output.
✅ Once enabled, the previous normal behavior for build cache applies.

(The only caveat being the message Ian pointed out above, where it points you at the wrong file if you run `update-cloud-credentials` and the build cache is disabled.)

File: apps/rush-lib/src/cli/actions/UpdateCloudCredentialsAction.ts (hunk 1 above)
Comment (octogonz) [reply apps/rush-lib/src/cli/actions/UpdateCloudCredentialsAction.ts:59]: `loadFromDefaultPathAsync()` claims that:
```
* If the file has not been created yet, then undefined is returned.
```

**Pull request merged**